feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c

```diff
@@ -980,119 +1006,172 @@ void vif_hs20_update(struct schema_Hotspot20_Config *hs2conf)
 	}
 }
 
+/* Mesh options table */
+#define VIF_SCHEMA_MESH_OPT_SZ            32
+#define VIF_SCHEMA_MESH_OPTS_MAX          2
+
+const char vif_mesh_options_table[VIF_SCHEMA_MESH_OPTS_MAX][VIF_SCHEMA_MESH_OPT_SZ] =
+{
+		SCHEMA_CONSTS_MESH_MCAST_RATE,
+		SCHEMA_CONSTS_MESH_MTU,
+};
+
+static void vif_mesh_opt_set(struct blob_buf *b, struct blob_buf *b1,
+		const struct schema_Wifi_VIF_Config *vconf)
+{
+	int i;
+	const char *opt;
+	const char *val;
+
+	for (i = 0; i < VIF_SCHEMA_MESH_OPTS_MAX; i++) {
+		opt = vif_mesh_options_table[i];
+		val = SCHEMA_KEY_VAL_NULL(vconf->mesh_options, opt);
+
+		if (strcmp(opt, "mcast_rate") == 0) {
+			if (!val)
+				blobmsg_add_u32(b, "mcast_rate", 24000);
+			else
+				blobmsg_add_u32(b, "mcast_rate", atoi(val));
+		}
+		else if (strcmp(opt, "mtu") == 0) {
+			if (!val)
+				blobmsg_add_u32(b1, "mtu", 1532);
+			else
+				blobmsg_add_u32(b1, "mtu", atoi(val));
+		}
+	}
+}
+
 bool target_vif_config_set2(const struct schema_Wifi_VIF_Config *vconf,
-			    const struct schema_Wifi_Radio_Config *rconf,
-			    const struct schema_Wifi_Credential_Config *cconfs,
-			    const struct schema_Wifi_VIF_Config_flags *changed,
-			    int num_cconfs)
+		const struct schema_Wifi_Radio_Config *rconf,
+		const struct schema_Wifi_Credential_Config *cconfs,
+		const struct schema_Wifi_VIF_Config_flags *changed,
+		int num_cconfs)
 {
 	int vid = 0;
 
 	blob_buf_init(&b, 0);
+	blob_buf_init(&mesh, 0);
 
 	blobmsg_add_string(&b, "ifname", vconf->if_name);
 	blobmsg_add_string(&b, "device", rconf->if_name);
-	blobmsg_add_string(&b, "mode", "ap");
 
 	if (changed->enabled)
 		blobmsg_add_bool(&b, "disabled", vconf->enabled ? 0 : 1);
 
-	if (changed->ssid)
-		blobmsg_add_string(&b, "ssid", vconf->ssid);
+	vif_config_security_set(&b, vconf);
 
-	if (changed->ssid_broadcast) {
-		if (!strcmp(vconf->ssid_broadcast, "disabled"))
-			blobmsg_add_bool(&b, "hidden", 1);
-		else
-			blobmsg_add_bool(&b, "hidden", 0);
-	}
+	if (!strcmp(vconf->mode, "mesh")) {
```

Comment:
I am wondering if the code is more readable if we put this and the normal wifi setup each to their own function.

Reply:
Done